### PR TITLE
DOC/MAINT: add references to Boost

### PR DESCRIPTION
**Pull request by dschmitz89**
#### What does this implement/fix?
Although many special functions used in `special` or `stats` rely on boost, we did not add references to boost yet. I personally find it important to credit the authors of the code we use.

#### Additional information
There are also many referenes left to the Fortran implementations of specfun, amos and cdflib. I would alter the descriptions along "reimplementation of the methods described in .." but will leave that for discussion.

### CODE REVIEW

**Review by steppi — CHANGES_REQUESTED** [review]
Thanks @dschmitz89. This is a welcome update. I agree we should be crediting Boost here. Rather than just adding a reference, I think it would be good to also add sentences “Wrapper for the Boost Math [i]_ routine X”.  If convenient, it would be good to be more specific with the references, but I don’t think it’s necessary for you to do that here. There’s still an issue of having to link to specific versions and keeping the versions up to date, and just adding any reference is a big improvement.

**Comment by fancidev** [comment]
I think it’s a very good initiative! Not only does it gives proper credit, it also helps the (curious) user to locate the source of the implementation easily without having to dig into the code. In this regard, I think the initiative equally applies to cdflib, cephes, etc.

**Comment by dschmitz89** [comment]
> > I think it’s a very good initiative! Not only does it gives proper credit, it also helps the (curious) user to locate the source of the implementation easily without having to dig into the code. In this regard, I think the initiative equally applies to cdflib, cephes, etc.
> 
> I think most special function docstrings do reference the original libraries when relevant. Boost is an exception because we’ve only started wrapping Boost relatively recently and haven’t updated the docstrings yet. 
> 
> What @dschmitz89 is talking about is changing wording like “wrapper for routine X from Zhang and Jin” to “Reimplemented based on routine X from Zhang and Jin.” Although in some cases the code has diverged from the original enough that it could be considered a new implementation. Basically every scalar kernel in `scipy.special` except those that come from Boost are being reimplemented.

Yap, let's stick to boost instead here. Too many moving parts in the rest of special. How about we use this as a reminder that when we touch a function again next time we try to keep the docstring in sync? Doc updates are easily forgotten, have been guilty of it myself.

**Comment by dschmitz89** [comment]
I cancelled unnecessary docs irrelevant jobs, thats why Github is showin all red CI.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -9454,15 +9504,19 @@ class skewnorm_gen(rv_continuous):
     When ``a = 0`` the distribution is identical to a normal distribution
     (`norm`). `rvs` implements the method of [1]_.
 
-    %(after_notes)s
+    This distribution uses routines from the Boost C++ library for
+    the computation of ``cdf``, ``ppf`` and ``isf`` methods. [1]_
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:9508]:
```suggestion
    the computation of ``cdf``, ``ppf`` and ``isf`` methods. [2]_
```

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -1046,10 +1052,15 @@ def add_newdoc(name, doc):
     betainc : regularized incomplete beta function
     gamma : gamma function
 
+    Notes
+    -----
+    This function wraps the ``ibeta_inv`` routine from the C++ Boost library [2]_.
```
Comment (steppi) [inline scipy/special/_add_newdocs.py:1057]:
```suggestion
    This function wraps the ``ibeta_inv`` routine from the Boost Math C++ library [2]_.
```

and similarly in the other docstrings. Boost is a set of libraries, not a single library, and the particular library used here is [Boost Math](https://github.com/boostorg/math).

**Comment by mdhaber** [comment]
There may be a few more hairs to be split (e.g.. precise placement of citations in the distribution notes), and I think this will be challenging to keep up to date, but probably better overall to merge and make changes as the need is noticed. Thanks @dschmitz89!

**Pull request merged**